### PR TITLE
upgrade rocksdb 5.4.5

### PR DESCRIPTION
**Pull request by siddontang**
PTAL @BusyJay @zhangjinpeng1987

### CODE REVIEW

**Comment by BusyJay** [comment]
ci fail

**Comment by siddontang** [comment]
should wait https://github.com/pingcap/rust-rocksdb/pull/61

File: librocksdb_sys/crocksdb/rocksdb/c.h [hunk 1]
```diff
@@ -792,8 +793,8 @@ extern C_ROCKSDB_LIBRARY_API void crocksdb_options_set_report_bg_io_stats(
     crocksdb_options_t*, int);
 extern C_ROCKSDB_LIBRARY_API void crocksdb_options_set_compaction_readahead_size(
     crocksdb_options_t*, size_t);
-extern C_ROCKSDB_LIBRARY_API void crocksdb_options_set_max_subcompactions(
-    crocksdb_options_t*, size_t);
+extern C_ROCKSDB_LIBRARY_API void
+crocksdb_options_set_max_subcompactions(crocksdb_options_t *, uint32_t);
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: space

File: librocksdb_sys/crocksdb/rocksdb/c.h [hunk 2]
```diff
@@ -404,8 +404,8 @@ extern C_ROCKSDB_LIBRARY_API void crocksdb_iter_get_error(
 /* Write batch */
 
 extern C_ROCKSDB_LIBRARY_API crocksdb_writebatch_t* crocksdb_writebatch_create();
-extern C_ROCKSDB_LIBRARY_API crocksdb_writebatch_t*
-    crocksdb_writebatch_create_with_capacity(size_t reserved_bytes);
+extern C_ROCKSDB_LIBRARY_API crocksdb_writebatch_t *
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: space

File: librocksdb_sys/crocksdb/rocksdb/c.h (hunk 1 above)
Comment (siddontang) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: This is formatted by RocksDB format style.

File: librocksdb_sys/crocksdb/rocksdb/c.h (hunk 2 above)
Comment (siddontang) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: This is formatted by RocksDB format style.

**Comment by siddontang** [comment]
PTAL @BusyJay @zhangjinpeng1987

File: librocksdb_sys/crocksdb/rocksdb/c.h (hunk 2 above)
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: But it is not consistency with upper and lower codes.

Comment (siddontang) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: `make format` does it automatically, so let us keep it here.

Comment (siddontang) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: I will see later how to adjust the format style.

File: librocksdb_sys/crocksdb/rocksdb/c.h (hunk 1 above)
Comment (andelf) [inline librocksdb_sys/crocksdb/rocksdb/c.h]: space between `crocksdb_options_t` and `*`

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**